### PR TITLE
[Launcher] Unit tests and fixes

### PR DESCRIPTION
**Pull request by alonmr**
Server side and more in-depth tests are in progress

### CODE REVIEW

File: mlrun/launcher/local.py [hunk 1]
```diff
@@ -216,14 +216,15 @@ def _create_local_function_for_execution(
 
         meta = mlrun.model.BaseMetadata(function_name, project=project)
 
-        command, runtime = mlrun.run.load_func_code(
+        command, _runtime = mlrun.run.load_func_code(
```
Comment (liranbg) [inline mlrun/launcher/local.py:219]: why tho you need `runtime` and `_runtime` scoped within same function? it is pretty error-prone

File: mlrun/launcher/local.py [hunk 2]
```diff
@@ -199,13 +199,13 @@ def execute(
...
-        name,
-        workdir,
-        handler,
+        runtime: "mlrun.runtimes.BaseRuntime",
+        run: "mlrun.run.RunObject",
+        local_code_path: Optional[str] = None,
+        project: Optional[str] = "",
+        name: Optional[str] = "",
+        workdir: Optional[str] = "",
+        handler: Optional[str] = None,
```
Comment (liranbg) [inline mlrun/launcher/local.py:208]: 🏅

File: tests/launcher/test_factory.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+@pytest.mark.parametrize(
+    "is_remote, local, expectation",
+    [
+        (True, False, does_not_raise()),
+        (False, False, does_not_raise()),
+        (True, True, pytest.raises(mlrun.errors.MLRunInternalServerError)),
+        (False, True, pytest.raises(mlrun.errors.MLRunInternalServerError)),
+    ],
+)
+def test_create_server_side_launcher(is_remote, local, expectation):
```
Comment (liranbg) [inline tests/launcher/test_factory.py:65]: not sure i understand the value of such test

**Review by liranbg — CHANGES_REQUESTED** [review]
minor

File: mlrun/launcher/local.py (hunk 1 above)
Comment (alonmr) [reply mlrun/launcher/local.py:219]: fixed, added a comment to explain what I did

File: tests/launcher/test_factory.py (hunk 3 above)
Comment (alonmr) [reply tests/launcher/test_factory.py:65]: just to see that if `is_running_as_api` is true then the factory creates the server side launcher and that it raises an error if local is true as well

**Review by Tankilevitch — APPROVED** [review]
LGTM,
Personally I would add a comment for each case in the `test_factory.py` so we will have understanding on why we expect each behavior while its still fresh for us.

**Pull request merged**